uniffi_bindgen/src/interface/mod.rs

```diff
@@ -1039,6 +1070,84 @@ impl APIConverter<Error> for weedle::EnumDefinition<'_> {
     }
 }
 
+#[derive(Debug, Clone)]
+pub struct CallbackInterface {
+    name: String,
+    methods: Vec<Method>,
+    ffi_init_callback: FFIFunction,
+}
+
+impl CallbackInterface {
+    fn new(name: String) -> CallbackInterface {
+        CallbackInterface {
+            name,
+            methods: Default::default(),
+            ffi_init_callback: Default::default(),
+        }
+    }
+
+    pub fn name(&self) -> &str {
+        &self.name
+    }
+
+    pub fn methods(&self) -> Vec<&Method> {
+        self.methods.iter().collect()
+    }
+
+    pub fn ffi_init_callback(&self) -> &FFIFunction {
+        &self.ffi_init_callback
+    }
+
+    fn derive_ffi_funcs(&mut self, ci_prefix: &str) -> Result<()> {
+        self.ffi_init_callback.name = format!("ffi_{}_{}_init_callback", ci_prefix, self.name);
+        self.ffi_init_callback.arguments = vec![FFIArgument {
+            name: "callback_stub".to_string(),
+            type_: FFIType::ForeignCallback,
+        }];
+        self.ffi_init_callback.return_type = None;
+        Ok(())
+    }
+}
+
+impl Hash for CallbackInterface {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        // We don't include the FFIFunc in the hash calculation, because:
+        //  - it is entirely determined by the other fields,
+        //    so excluding it is safe.
+        //  - its `name` property includes a checksum derived from  the very
+        //    hash value we're trying to calculate here, so excluding it
+        //    avoids a weird circular depenendency in the calculation.
+        self.name.hash(state);
+        self.methods.hash(state);
+    }
+}
+
+impl APIConverter<CallbackInterface> for weedle::CallbackInterfaceDefinition<'_> {
+    fn convert(&self, ci: &mut ComponentInterface) -> Result<CallbackInterface> {
+        if self.attributes.is_some() {
+            bail!("interface attributes are not supported yet");
```

Comment:
nit: please say "callback interface" rather than just "interface" in these error strings.

Reply:
Yes. This was a bit of copy pasta.